src/utils/SocketClient.cxxtest

```diff
@@ -162,3 +164,27 @@ TEST_F(SocketClientTest, DISABLED_connect_mdns_not_published)
     delete sl;
     delete sc;
 }
+
+TEST_F(SocketClientTest, connect_host_disallow_local)
+{
+    SocketListener *sl = new SocketListener(LISTEN_PORT, accept_callback);
+
+    SocketClient *sc = new SocketClient(
+        node_->iface()->dispatcher()->service(), nullptr, "127.0.0.1",
+        LISTEN_PORT,
+        std::bind(&SocketClientTest::connect_callback, this, _1, _2, _3),
+        std::bind(&SocketClientTest::status_callback, this, _1), 1, 1, true);
+
+    EXPECT_CALL(*this, connect_callback(_, _, sc)).Times(0);
+    EXPECT_CALL(*this, status_callback(SocketClient::Status::CONNECT_FAILED_SELF)).Times(AtLeast(1));
+
+    usleep(40000);
+    wait();
+    usleep(40000);
+    wait();
+
+    sl->shutdown();
+    sc->shutdown();
+    delete sl;
```

Comment:
if you allocate these within the funciton then delete them before returning, it's easier to allocate them on the stack as a local variable.